src/main/java/converters/BytesConverter.java

```diff
@@ -0,0 +1,86 @@
+package converters;
+
+import exceptions.ConverterException;
+import options.IOption;
+
+/**
+ * This class provides methods for converting between strings of bytes and binary strings.
+ * It extends the abstract {@link Converter} class.
+ * <p>
+ * To convert a string of bytes to a binary string, use the {@link #convertTo(String, IOption[])} method.
+ * To convert a binary string to a string of bytes, use the {@link #convertFrom(String, IOption[])} method.
+ * <p>
+ * The BytesConverter doesn't support any options
+ */
+public class BytesConverter extends Converter {
+
+	/**
+	 * Converts a binary string to a string of bytes.
+	 *
+	 * @param bitStr  the binary string to convert
+	 * @param options an array of {@link IOption} options (options are ignored)
+	 * @return the string of bytes
+	 * @throws ConverterException if the input binary string is invalid
+	 */
+	@Override
+	public String convertTo(String bitStr, IOption[] options) throws ConverterException {
+		return this.convertTo(bitStr);
+	}
+
+	/**
+	 * Converts a binary string to a string of bytes.
+	 *
+	 * @param bitStr  the binary string to convert
+	 * @return the string of bytes
+	 * @throws ConverterException if the input binary string is invalid
+	 */
+	public String convertTo(String bitStr) throws ConverterException {
+		validateInput(bitStr, "^[01]+$");
```

Comment:
**thought, issue**: bit string internal representation causes inefficient and cumbersome code in other converters

The code could be more efficient if you had used byte representation instead.
I have not done the math but for every conversion you are allocating at least 8x more memory than your program could have used.

Reply:
**note:** according to the Intellij profiler, you are using circa 800 MB of memory to digest a 50 MB file. Format (-f int -t int).

Reply:
issue: Hence the conversion to the bit string, you have mentioned a limitation in the readme about parsing non-UTF-8 symbols.

> Our application does not support converting integers bigger than 1B to bytes. The application will not crash or print errors but convert integer byte after byte. This means that non-UTF-8 characters will not be converted as 1 character but rather as a sequence of multiple UTF-8 characters.

But that's not true. Your application breaks after extended ASCII, which means symbols with positions bigger than 255 are not correctly parsed, although they are still within the UTF-8 range.